frontend/components/admin.jsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import { render } from 'react-dom';
+import io from 'socket.io-client';
+class Admin extends React.Component {
+  constructor(props) {
+    super(props);
+    const socket = io(SOCKET_URL);
+    socket.on('admin_return', (data) => {
+      var dataArray = []
+
+      this.setState({gamestate : data['gamestate'], tabledata : data['tabledata']});
+    console.log('i heard an answer')
+    console.log(this.state)
+    });
+    this.state = {
+      socket:socket,
+      tabledata : ['initialvalue'],
+      gamestate : {},
+      info: [['Hello! Welcome to the Admin Page. Press the button to see the progress!']],
+    };
+    this.adminPing();
+  }
+
+  adminPing() {
+    const { socket } = this.state
+    socket.emit('admin_ping', {})
+    console.log('i made a ping')
+  }
+
+  renderAdminfunction() {
+    const { tabledata } = this.state
+    console.log('testerfunction here')
+    let tableelements
+    tableelements = Object.entries(tabledata).map(elements => {
+      var table = elements[0]
+      console.log(table)
+      var clientlist = this.renderClients(table);
+      var tablegamestate = this.renderPuzzle(table);
+       return (
+        <div key={table}>
+          <h1>{table}</h1>
+          <p>Participants: {clientlist} </p>
+          <p>Status: {tablegamestate} </p>
+        </div>
+      );
+    })
+     this.setState({info:tableelements})
```

Comment:
storing JSX in state is a bit strange, can we move it into the render method instead?